### PR TITLE
Updated gradient_clip_val and gradient_clip_algorithm default, batch size comment

### PR DESCRIPTION
**Pull request by bhavyak7-lab**
Changes made:
- Updated gradient_clip_val and gradient_clip_algorithm default 
- Updated batch size comment 
- Updated changelog
- Upgraded pytorch lightning version

## Summary by CodeRabbit

* **Dependencies**
  * Minimum Lightning requirement raised to 2.6
  * Minimum Python requirement raised to 3.10

* **Changed**
  * Gradient clipping defaults set: gradient_clip_val = 1.0 and gradient_clip_algorithm = "norm"
  * Batch size documentation clarified to explain per-device micro-batch sizing and effective optimization batch size in multi-GPU setups

* **Tests**
  * Updated expected de novo prediction strings and scan identifier format in tests

* **Changelog**
  * Release notes updated to reflect the above defaults and documentation clarifications

### CODE REVIEW

File: tests/test_integration.py [hunk 1]
```diff
@@ -80,7 +80,7 @@ def test_train_and_run(
     # Verify that the spectrum predictions are correct and indexed
     # according to the peak input file type.
     psms = mztab.spectrum_match_table
-    assert psms.loc[1, "sequence"] == "LESLLEK"
```
Comment (bhavyak7-lab) [inline tests/test_integration.py:83]: For this test, it passes with the original peptide when I run it locally, but is not passing when I run the tests online. I have been looking into why, but I think it is because of a version of a package that was updated.

**Comment by bhavyak7-lab** [comment]
The code coverage test should be resolved by PR #1191
